File: Leetcode 5. Longest Palindromic Substring/lc 5.cpp
class Solution {
public:
    string findPdrm(string s,int l,int r){
        while(l>-1 && r<s.length() && s[l]==s[r]){
            l--;
            r++;
        }
        return s.substr(l+1,r-l-1);
    }
    string longestPalindrome(string s) {
        string ans="";
        for(int i=0;i<s.length();i++){
            //find for odd
            string s1=findPdrm(s,i,i);
            //find for even
            string s2=findPdrm(s,i,i+1);

            if(ans.length()<max(s1.length(),s2.length())){
                if(s1.length()>s2.length()){
                    ans=s1;
                }else{
                    ans=s2;
                }
            }
        }
        return ans;
    }
};

File: Leetcode 5. Longest Palindromic Substring/lc 5.java
class Solution {
    public String longestPalindrome(String s) {
        int len=s.length();

        while(len>0){
            for(int i=0;i<s.length()-len+1;i++){
                if(check(s.substring(i,i+len))){
                    return s.substring(i,i+len);
                }
            }
            len--;
        }
        return "";
    }
    boolean check(String s){
        int start=0;
        int end=s.length()-1;
        while(start<=end){
            if(s.charAt(start)==s.charAt(end)){
                start++;
                end--;
            }else{
                return false;
            }
        }
        return true;
    }
}